types/src/main.ts

```diff
@@ -4087,6 +4087,144 @@ export namespace InlayHint {
 	}
 }
 
+/**
+ * A snippet string is a template which allows to insert text
+ * and to control the editor cursor when insertion happens.
+ *
+ * A snippet can define tab stops and placeholders with `$1`, `$2`
+ * and `${3:foo}`. `$0` defines the final tab stop, it defaults to
+ * the end of the snippet. Variables are defined with `$name` and
+ * `${name:default value}`.
+ *
+ * @since 3.18.0
+ */
+export interface SnippetString {
```

Comment:
To make this easier to extend for the future we should have something more like 

```
export interface StringValue {
	kind: 'snippet';
	value: string;
}
```

Otherwise it is very hard to evolve in the future.

Reply:
Done.